UPGRADE.md

```diff
@@ -39,11 +41,11 @@ New rule:
 > * Characters start with an alphabetic character.
 > * Characters end with an alphanumeric character.
 
-### MeshService on Universal is not generated for services with an invalid `kuma.io/service`
+### On Universal, a MeshService is no longer generated for any dataplane that has an invalid `kuma.io/service` tag
 
-When using `MeshService`, we were automatically generating `MeshService` resources in Universal mode. However, due to stricter resource validation, we have decided not to generate `MeshService` for dataplanes with a `kuma.io/service` name that is not [RFC 1035](https://www.rfc-editor.org/rfc/rfc1035.html) compliant.
+When using `MeshService`, we were automatically generating `MeshService` resources in Universal mode. However, due to stricter resource validation, we have decided not to generate the `MeshService` object when a dataplane has a `kuma.io/service` that is not [RFC 1035](https://www.rfc-editor.org/rfc/rfc1035.html) compliant.
 
-If you want `MeshService` to be autogenerated by the control plane on Universal, update the `kuma.io/service` tag to a valid name. Otherwise, you must create the `MeshService` manually.
+If you want `MeshService` objects to be autogenerated by the control plane on Universal, update the `kuma.io/service` tags to valid names. Otherwise, you must create them manually.
```

Comment:
```suggestion
If you want the control plane on Universal to autogenerate `MeshService` resources, update the `kuma.io/service` tags to valid names. Otherwise, you must create them manually.
```